src/sentry/static/sentry/app/components/events/interfaces/breadcrumbsV2/breadcrumbFilter/breadcrumbFilter.tsx

```diff
@@ -0,0 +1,161 @@
+import React from 'react';
+import styled from '@emotion/styled';
+import isEqual from 'lodash/isEqual';
+
+import {t} from 'app/locale';
+import DropdownControl from 'app/components/dropdownControl';
+import DropdownButton from 'app/components/dropdownButton';
+
+import BreadcrumbFilterGroup from './breadcrumbFilterGroup';
+import BreadcrumbFilterHeader from './breadcrumbFilterHeader';
+import BreadcrumbFilterFooter from './breadcrumbFilterFooter';
+import {FilterGroup, FilterGroupType, FilterType} from './types';
+
+type Props = {
+  onFilter: (filterGroups: Array<FilterGroup>) => () => void;
+  filterGroups: Array<FilterGroup>;
+};
+
+type State = {
+  filterGroups: Array<FilterGroup>;
+};
+
+class BreadcrumbFilter extends React.Component<Props, State> {
+  state: State = {
+    filterGroups: [],
+  };
+
+  componentDidUpdate(prevProps: Props) {
+    if (!isEqual(this.props.filterGroups, prevProps.filterGroups)) {
+      this.loadState();
+    }
+  }
+
+  loadState() {
+    const {filterGroups} = this.props;
+
+    this.setState({
+      filterGroups,
+    });
+  }
+
+  handleClickItem = (type: FilterType, groupType: FilterGroupType) => {
+    const {filterGroups} = this.state;
+    const newFilterGroups = filterGroups.map(filterGroup => {
+      if (filterGroup.groupType === groupType && filterGroup.type === type) {
+        return {
+          ...filterGroup,
+          isChecked: !filterGroup.isChecked,
+        };
+      }
+      return filterGroup;
+    });
+
+    this.setState({
+      filterGroups: newFilterGroups,
+    });
+  };
+
+  handleSelectAll = (selectAll: boolean) => {
+    this.setState(prevState => ({
+      filterGroups: prevState.filterGroups.map(data => ({
+        ...data,
+        isChecked: selectAll,
+      })),
+    }));
+  };
+
+  getDropDownButton = ({isOpen, getActorProps}) => {
+    const {filterGroups} = this.props;
+
+    const checkedFilterGroupsOptionsQuantity = filterGroups.filter(
+      filterGroup => filterGroup.isChecked
+    ).length;
+
+    let buttonLabel = t('Filter By');
+
+    if (checkedFilterGroupsOptionsQuantity > 0) {
+      buttonLabel = `${checkedFilterGroupsOptionsQuantity} ${
+        checkedFilterGroupsOptionsQuantity === 1
+          ? t('Active Filter')
+          : t('Active Filters')
```

Comment:
You should use tn() for this. Pluralization rules vary by locale.

Reply:
I didn't know about it...thanks for the tip 😉